### PR TITLE
Viite 685 reserved road part table

### PR DESCRIPTION
**Pull request by tvenhola**
Major refactor on reserving road parts for project.

### CODE REVIEW

File: digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala [hunk 1]
```diff
@@ -207,9 +209,12 @@ class ViiteApi(val roadLinkService: RoadLinkService, val vVHClient: VVHClient,
...
     try {
-      val (projectSaved, addr, info, success) = projectService.createRoadLinkProject(roadAddressProject)
-      Map("project" -> roadAddressProjectToApi(projectSaved), "projectAddresses" -> addr, "formInfo" -> info,
-        "success" -> success)
+      val projectSaved = projectService.createRoadLinkProject(roadAddressProject)
+      val fetched = projectService.getRoadAddressSingleProject(projectSaved.id).get
+      val firstAddress: Map[String, Any] =
+        fetched.reservedParts.find(_.startingLinkId.nonEmpty).map(p => "projectAddresses" -> p.startingLinkId.get).toMap
+      Map("project" -> roadAddressProjectToApi(fetched), "formInfo" ->
+        fetched.reservedParts.map(reservedRoadPartToApi), "success" -> true) ++ firstAddress
```
Comment (Campanha) [inline digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala]: do we need "trPreview" as in line 239?

File: digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/dao/ProjectLinkDaoSpec.scala [hunk 2]
```diff
@@ -45,21 +45,11 @@ class ProjectLinkDaoSpec  extends FunSuite with Matchers {
...
-      addprojects()
-      /*Insert links to project*/
-      sqlu"""insert into project_link (id,project_id,track_code,discontinuity_type,road_number,road_part_number,start_addr_M,end_addr_M,lrm_position_id,created_by) VALUES (1,1,1,0,1,1,1,1,20000286,'automatedtest')""".execute
-      sqlu"""insert into project_link (id,project_id,track_code,discontinuity_type,road_number,road_part_number,start_addr_M,end_addr_M,lrm_position_id,created_by) VALUES (2,2,1,0,1,1,1,1,20000287,'automatedtest')""".execute
-      sql"""SELECT COUNT(*) FROM project_link WHERE created_by = 'automatedtest'""".as[Long].first should be(2L)
-      dynamicSession.rollback()
-    }
-  }
-
   //TRIGGER has been removed for the time being. Should activate the test again if the triggers are activated again.
```
Comment (oappicgi) [inline digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/dao/ProjectLinkDaoSpec.scala]: should this comment be removed if test is no longer ignored?

Comment (tvenhola) [reply digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/dao/ProjectLinkDaoSpec.scala]: Yes

File: digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala (hunk 1 above)
Comment (tvenhola) [reply digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala]: I think it's used to refresh the change table view. Or should be.

Comment (tvenhola) [reply digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala]: Well, not after saving the project. I'll remove that change table and it'll be faster, too.

**Pull request merged**